changelogs/unreleased/gh-9723-sync-system-spaces.md

```diff
@@ -0,0 +1,9 @@
+## feature/replication
+
+* Made all of the system spaces synchronous by default when the synchronous
+  queue is claimed. A `box_consider_system_spaces_synchronous` backward
```

Comment:
The issue, the documentation request and the changelog entry define a predicate when the system spaces become synchronous this way:

> when the synchronous queue is claimed

It is obscure from a user perspective. I think that the documentation request should shed some light here.

I asked @sergepetrenko and if I understood his words correctly, whether the system spaces are synchronous depends on the following properties:

* The compat option.
* Whether RAFT is in use.
* Whether `box.ctl.promote()` and `box.ctl.demote()` is used and which of them is the last.

And these properties are somehow related to whether it is possible that synchronous transactions are present in the replicaset history. I think that it is important to highlight this relation.

Reply:
I don't think it is obscure. The predicate is simply `box.info.synchro.queue.owner ~= 0`, and it is described in the documentation [^1]:

> `owner` (since version [2.10.0](https://www.tarantool.io/ru/doc/latest/release/2.10.0/)) – ID of the replica that owns the synchronous transaction queue. Once an owner instance appears, all other instances become read-only. If the owner field is 0, then every instance may be writable, but they can’t create any synchronous transactions. To claim or reclaim the queue, use [box.ctl.promote()](https://www.tarantool.io/ru/doc/latest/reference/reference_lua/box_ctl/promote/#box-ctl-promote) on the instance that you want to promote. With elections enabled, an instance runs box.ctl.promote() command automatically after winning the elections. To clear the ownership, call [box.ctl.demote()](https://www.tarantool.io/ru/doc/latest/reference/reference_lua/box_ctl/demote/#box-ctl-demote) on the synchronous queue owner.

[^1]: https://www.tarantool.io/ru/doc/latest/reference/reference_lua/box_info/synchro/

Reply:
I agree with @CuriousGeorgiy here. We have mentioned `box.info.synchro.queue.owner` on multiple occasions in the documentation as being a predicate for some mode of operation.

I also agree that the documentation should be clarified regarding the cases when `synchro.queue.owner` is not nil. I've opened a corresponding doc ticket: https://github.com/tarantool/doc/issues/4400

Reply:
Resolving since the PR is approved.